substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrManager.java

```diff
@@ -0,0 +1,255 @@
+    }
+
+    private static Long parseDuration(Map<JfrStartArgument, String> args, JfrStartArgument key) {
+        String value = args.get(key);
+        if (value != null) {
+            try {
+                int idx = indexOfFirstNonDigitCharacter(value);
+                long time;
+                try {
+                    time = Long.valueOf(value.substring(0, idx));
+                } catch (IllegalArgumentException e) {
+                    throw new IllegalArgumentException("Expected a number.");
+                }
+
+                if (idx == value.length()) {
+                    // only accept missing unit if the value is 0
+                    if (time != 0) {
+                        throw new IllegalArgumentException("Unit is required.");
+                    }
+                    return 0L;
+                }
+
+                String unit = value.substring(idx);
+                if ("ns".equals(unit)) {
+                    return Duration.ofNanos(time).toNanos();
+                } else if ("us".equals(unit)) {
+                    return Duration.ofNanos(time * 1000).toNanos();
+                } else if ("ms".equals(unit)) {
+                    return Duration.ofMillis(time).toNanos();
+                } else if ("s".equals(unit)) {
+                    return Duration.ofSeconds(time).toNanos();
+                } else if ("m".equals(unit)) {
+                    return Duration.ofMinutes(time).toNanos();
+                } else if ("h".equals(unit)) {
+                    return Duration.ofHours(time).toNanos();
+                } else if ("f".equals(unit)) {
+                    return Duration.ofDays(time).toNanos();
+                }
+                throw new IllegalArgumentException("Unit is invalid.");
+            } catch (IllegalArgumentException e) {
+                throw UserError.abort("Could not parse JFR argument '" + key.cmdLineKey + "=" + value + "'. " + e.getMessage());
+            }
+        }
+        return null;
+    }
+
+    private static int indexOfFirstNonDigitCharacter(String durationText) {
+        int idx = 0;
+        while (idx < durationText.length() && Character.isDigit(durationText.charAt(idx))) {
+            idx++;
+        }
+        return idx;
+    }
+
+    private static JfrStartArgument findArgument(JfrStartArgument[] possibleArguments, String value) {
+        for (JfrStartArgument arg : possibleArguments) {
+            if (arg.cmdLineKey.equals(value)) {
+                return arg;
+            }
+        }
+        return null;
+    }
+
+    public static class Options {
+        @Option(help = "Enable Java Flight Recorder.")//
+        public static final RuntimeOptionKey<Boolean> FlightRecorder = new RuntimeOptionKey<>(false);
+
+        @Option(help = "Start flight recording with options.")//
+        public static final RuntimeOptionKey<String> StartFlightRecording = new RuntimeOptionKey<>("");
+    }
+
+    private enum JfrStartArgument {
+        Name("name"),
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.